sklearn/decomposition/_nmf.py

```diff
@@ -1183,6 +1201,19 @@ def _check_params(self, X):
         self._beta_loss = _beta_loss_to_float(self.beta_loss)
 
     def _check_w_h(self, X, W, H, update_H):
+        # n_components
+        if self._n_components == "auto":
+            if W is not None:
+                if H is not None:
+                    self._n_components = H.shape[0]
+                else:
+                    self._n_components = W.shape[1]
```

Comment:
I don't think this case can happen. We can have either W and H provided or only H provided but not only W